digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/lane/LaneDao.scala

```diff
@@ -0,0 +1,454 @@
+               la.name, la.value, l.municipality_code, l.lane_code
+          FROM lane l
+          JOIN lane_link ll ON l.id = ll.lane_id
+          JOIN lane_position pos ON ll.lane_position_id = pos.id
+          JOIN lane_attribute la ON la.lane_id = l.id
+          JOIN #$idTableName i ON i.id = pos.link_id
+          WHERE (l.valid_to > sysdate OR l.valid_to IS NULL)
+          AND pos.side_code = $sideCode
+      """.as[LaneRow].list
+    }
+
+    convertLaneRowToPersistedLane (lanes)
+  }
+
+
+  /**
+    * Iterates a set of lanes ids. Used by LaneService.getPersistedLanesByIds,
+    * laneService.split and laneService.separate.
+    */
+  def fetchLanesByIds(ids: Set[Long] ): Seq[PersistedLane] = {
+    val lanes = MassQuery.withIds(ids) { idTableName =>
+      sql"""SELECT l.id, pos.link_id, pos.side_code, pos.start_measure, pos.end_measure,
+               l.created_by, l.created_date, l.modified_by, l.modified_date,
+               CASE WHEN l.valid_to <= sysdate THEN 1 ELSE 0 END AS expired,
+               pos.adjusted_timestamp, pos.modified_date,
+               la.name, la.value, l.municipality_code, l.lane_code
+          FROM lane l
+          JOIN lane_link ll ON l.id = ll.lane_id
+          JOIN lane_position pos ON ll.lane_position_id = pos.id
+          JOIN lane_attribute la ON la.lane_id = l.id
+          JOIN #$idTableName i ON i.id = l.id
+      """.as[LaneRow].list
+    }
+
+    convertLaneRowToPersistedLane (lanes)
+
+  }
+
+
+  def convertLaneRowToPersistedLane(lanes: Seq[LaneRow]): Seq[PersistedLane] = {
+    lanes.groupBy(_.id).map { case (id, assetRows) =>
+        val row = assetRows.head
+        val attributeValues = LanePropertiesValues(laneRowToProperty(assetRows))
+
+        id -> PersistedLane (id = row.id, linkId = row.linkId, sideCode = row.sideCode, laneCode = row.laneCode,
+          municipalityCode = row.municipalityCode, startMeasure = row.startMeasure, endMeasure = row.endMeasure,
+          createdBy = row.createdBy, createdDateTime = row.createdDate,
+          modifiedBy = row.modifiedBy, modifiedDateTime = row.modifiedDate,  expired = row.expired,
+          vvhTimeStamp = row.vvhTimeStamp, geomModifiedDate = row.geomModifiedDate, attributes = attributeValues)
+
+    }.values.toSeq
+  }
+
+
+  def laneRowToProperty( laneRows: Iterable[LaneRow]): Seq[LaneProperty] = {
+
+    val laneCodeAttribute = Seq(LaneProperty( "lane_code", Seq(LanePropertyValue(laneRows.head.laneCode))))
+
+    val props = laneRows.groupBy(_.value.publicId).map { case (key, rows) =>
+      val row = rows.head
+
+      LaneProperty(
+        publicId = row.value.publicId,
+
+        values = rows.flatMap(laneRow =>
+          laneRow.value.propertyValue match {
+            case Some(value) => Some(LanePropertyValue(value))
+            case _ => None
+          }
+        ).toSeq
+      )
+
+    }.toSeq
+
+    val finalAttributes = props ++ laneCodeAttribute
+    finalAttributes
```

Comment:
No need to use this val finalAttributes just put `props ++ laneCodeAttribute`

Reply:
Done.